app/src/main/java/io/neurolab/main/DeviceConnector.java

```diff
@@ -0,0 +1,121 @@
+package io.neurolab.main;
+
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbDeviceConnection;
+import android.hardware.usb.UsbManager;
+import android.util.Log;
+
+import com.felhr.usbserial.UsbSerialDevice;
+import com.felhr.usbserial.UsbSerialInterface;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DeviceConnector {
+    public final String ACTION_USB_PERMISSION = "com.hariharan.arduinousb.USB_PERMISSION";
```

Comment:
What is this part about; hariharan arduino?

Reply:
Ah! I was taking some inspiration from this guy's Arduino-Android code, he has some great stuff going on. Nevertheless, have corrected the perm constant now for us 👍